### ISSUE TITLE
UI broken when changing page while nuxt is hydrating

### ISSUE THREAD

**Issue by huang-julien** [nuxt/nuxt#21516]
### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.15.0
- Nuxt Version:     3.5.3
- Nitro Version:    2.4.1
- Package Manager:  pnpm@8.6.0
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

test fixtures

add `console.log(nuxtApp.isHydrating)` at https://github.com/nuxt/nuxt/blob/c6a62268c3f466054249e33c7db3ac37452cdf03/packages/nuxt/src/app/components/nuxt-island.ts#L105 to make reproduction easier

- go to `/`
- click immediatly on `island` to get redirected to `/islands`
- see the UI completly broken
![image](https://github.com/nuxt/nuxt/assets/63512348/0397e796-649c-40d8-bd6f-ef7d171cba6e)

### Describe the bug

Navigating while nuxt is hydrating can completly break the app. Aren't we blocking the navigation until the hydration process has finished with a nuxtlink ?

### Additional context

Found when debugging #21461, it can be one of the reasons the CI are failing but it's also reproductible on the main branch

edit: seems that this only happen with `/islands`

**Comment by itsmnthn** [comment]
IDK if I have the same issue. I'm not using any experimental feature, and fully static site generation and page transition are enabled. In production, rendering stops with an error when I switch between pages too quickly.

"Uncaught DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node."

**Comment by manniL** [comment]
@huang-julien Is this a duplicate of https://github.com/nuxt/nuxt/issues/13309 which also talks about breaking the app when navigating quickly? 🤔

**Comment by huang-julien** [comment]
@manniL i don't think so. This one happens at hydration, probably with server components. It's not the same issue but can be due to suspense too

**Comment by TheDutchCoder** [comment]
This is issue is a bit deeper rooted I think. We see a lot of hydration errors in DataDog, because components are already reactive while Nuxt is still hydrating.

This should, in theory, not be possible (components should only become active _after_ hydration is completed).

It's also fairly easy to reproduce, just really quickly keep clicking on a component (after a hard reload) that changes some state that's also rendered server side.

The example code I use is:
```ts
// Method in a Vue component
const selectTab = (tab: SearchTypes) => {
  console.log('selectTab', useNuxtApp().isHydrating)
}
```

Result in the browser (first few click are logged before hydration is completed (bad), last click occurred after (fine)):
<img width="595" alt="image" src="https://github.com/user-attachments/assets/eca4807b-85db-4fbc-8820-d65d628d619f">

**Comment by MirkoJa** [comment]
@huang-julien could you check if this is solved by https://github.com/nuxt/nuxt/pull/26480?

**Comment by danielroe** [comment]
I believe this was fixed by #26480 (shipped in v3.11.0)
